src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -731,6 +731,7 @@ protected bool TakeStartLine(SocketInput input)
                     // URI was encoded, unescape and then parse as utf8
                     pathEnd = UrlPathDecoder.Unescape(pathBegin, pathEnd);
                     requestUrlPath = pathBegin.GetUtf8String(pathEnd);
+                    requestUrlPath = PathNormalizer.NormalizeToNFC(requestUrlPath);
```

Comment:
Correcting

It currently does:
Utf8
1, `Unescape` - loop each byte 
2. `GetUtf8String` (`new char[byteLength * 2]`+ `new string()`) + loop each byte -> char
3. `NormalizeToNFC` string stuffs
4. `ContainsDotSegments` over string
5. `RemoveDotSegments`  (`ArrayPool<char>`+ `new string()`) + loop each char

Ascii
1. .GetAsciiString(pathEnd); `new string()`
2. `ContainsDotSegments` over string
3. `PathNormalizer.RemoveDotSegments`  `new string()`

Could do for the Utf8 route
1. `Unescape` - loop each byte 
2. `ContainsDotSegments` seek()
3. `RemoveDotSegments` loop each byte as per 1, (`.` and `\` are the same bytes in utf8)
4. `GetUtf8String` (`ArrayPool<char>`+ `new string()`)
5. `NormalizeToNFC` string stuffs 

And Ascii
1. `ContainsDotSegments` seek()
2. `RemoveDotSegments` `memoryblock`s (as per `Unescape`)
3. .GetAsciiString(pathEnd); `new string()`